src/components/YearSelector.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+interface YearSelectorProps {
+  selectedYear: number | null;
+  onYearChange: (year: number) => void;
+}
+
+const generateYears = (startYear: number, endYear: number) => {
+  const years = [];
+  for (let year = startYear; year <= endYear; year++) {
+    years.push(year);
+  }
+  return years;
+};
+
+const YearSelector: React.FC<YearSelectorProps> = ({ selectedYear, onYearChange }) => {
+  const years = generateYears(2023, 2030); //To change or add more or less years, change the start and end year here.
```

Comment:
rather then having the developer change the end year, this should be done automatically. typescript has a way to get the current year with datetime

Reply:
I have question in regards of this. I'll ask them in the meeting today.